### ISSUE TITLE
FR: stop going to next slide with variableWidth: true when last slide is visible

### ISSUE THREAD

**Issue by fritzmg** [kenwheeler/slick#3694]
See the following example: https://jsfiddle.net/Spooky_/s1475jpc/

If you keep going to the right, you can keep going until the last slide is at the left edge of the slider. However it doesn't really makes sense to go further than needed... there should be an option to stop being able to go right, if the last slide is fully visible.

**Comment by alimemonzx** [comment]
@fritzmg you have to add `slidesToShow: 4` in your slick function.

```
$('.slider').slick({
  variableWidth: true,
  infinite: false,
  slidesToShow: 4,
});
```
This will prevent the slider to go further and it disables the next button as well. Hope this helps

**Comment by fritzmg** [comment]
@alimemonzx that does not help, you cannot hard code the `slidesToShow`, otherwise the slider will slide further than necessary and skip slides.

**Comment by alimemonzx** [comment]
@fritzmg the behaviour which you are describing is done by `slidesToScroll: 4` the property `slidesToShow: 4` shows the number of slides in one div. If you want to show user one slide on one click you can use `slidesToScroll: 1`. By the way default is set to 1.

**Comment by fritzmg** [comment]
No, consider the following example: https://jsfiddle.net/e1q5fuam/1/

Setting `slidesToShow: …` together with `variableWidth: true` does not make any sense, since the width of each slide is variable.

**Comment by alimemonzx** [comment]
@fritzmg this example explains much better. You are right at your point. But the behaviour of current slides are next slide will appear after the previous is invisible.

So when slider goes on last slide it has to remove second last slide and show the last slide with an empty content.

Thanks for better example.

**Comment by fritzmg** [comment]
Yeah, you are right, I should have used the "true" variable width example in the first place :)

**Comment by vshepel** [comment]
@fritzmg Hi, I have a similar problem, did you solve it?

**Comment by fritzmg** [comment]
No, I have no solution.

**Comment by buccia** [comment]
I have the same problem!

**Comment by nsluc93** [comment]
I just have such a solution. hope it helps
// js 
$slick_slider.on('afterChange', function (e, slick) {
                    var lElRect = slick.$slides[slick.slideCount - 1].getBoundingClientRect();
                    var rOffset = lElRect.x + lElRect.width;
                    var wraRect = $('.slick-list', $slick_slider).get(0).getBoundingClientRect();
                    if (rOffset < (wraRect.x + wraRect.width)) {
                        $('.slick-next', $slick_slider).addClass('slick-disabled');
                    }
                });
// css 
.slick-disabled {
                        cursor: no-drop;
                        opacity: 0.5;
                        pointer-events: none;
                    }
